### PR TITLE
Descope OIDC and SAML Integration Guide

### PR DESCRIPTION
**Pull request by gaokevin1**
This PR adds guides on how to configure Descope as an OIDC and SAML provider for ngrok.

### CODE REVIEW

**Review by gshriki — APPROVED** [review]

**Comment by gaokevin1** [comment]
@akclaireshaw @russorat Is there any chance you can review this documentation and approve this?

**Comment by gaokevin1** [comment]
Hi @akclaireshaw @russorat any update on this?

File: docs/integrations/descope/index.mdx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+name: Descope
+title: Descope Integration Hub
+sidebar_label: Descope
+description: |
+  Secure access to ngrok tunnels with Descope.
```
Comment (russorat) [inline docs/integrations/descope/index.mdx:6]:
```suggestion
  Secure access to ngrok endpoints with Descope. 
```

File: docs/integrations/descope/index.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+name: Descope
+title: Descope Integration Hub
+sidebar_label: Descope
+description: |
+  Secure access to ngrok tunnels with Descope. 
+  No VPNs, Firewalls, or complex network rules required.
+excerpt: |
+  Secure access to your ngrok tunnels with Descope.
```
Comment (russorat) [inline docs/integrations/descope/index.mdx:9]:
```suggestion
  Secure access to your ngrok endpoints with Descope.
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 3]
```diff
@@ -0,0 +1,161 @@
+---
+description: Use Descope OIDC to secure access to ngrok tunnels
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:2]:
```suggestion
description: Use Descope OIDC to secure access to ngrok endpoints
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+---
+
+# Descope SSO (OpenID Connect)
+
+:::tip TL;DR
+
+To secure access to ngrok with Descope Single Sign-On using OpenID Connect:
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:9]:
```suggestion
To secure access to your ngrok endpoints with Descope Single Sign-On (SSO) using OpenID Connect (OIDC):
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+1. [Configure Descope SSO](#configure-Descope)
+1. [Configure ngrok](#configure-ngrok)
+1. [Test access to ngrok with Descope SSO](#test-sso)
+
+:::
+
+This article details how to configure Descope as the primary Identity Provider for ngrok tunnels.
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:17]:
```suggestion
This article details how to configure Descope as the primary Identity Provider for your ngrok endpoints.
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+- **Restrict access to ngrok tunnels** only to users authenticated via Descope
+- **Use Descope security policies and MFA authenticators**.
+- **Use the Descope Console to facilitate access to ngrok apps**.
+
+## Supported Features
+
+The ngrok integration with Descope supports:
+
+- **SP-Initiated SSO**: In this mode, users access ngrok edges and tunnels and are redirected to Descope for authentication.
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:28]:
```suggestion
- **SP-Initiated SSO**: In this mode, users access ngrok endpoints and are redirected to Descope for authentication.
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+## Requirements
+
+To configure ngrok tunnels with Descope, you must have:
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:32]:
```suggestion
To configure ngrok endpoints with Descope, you must have:
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+- an Descope account with administrative rights to a Descope Project
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:34]:
```suggestion
- A Descope account with administrative rights to a Descope Project
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+- an ngrok enterprise account with an access token or admin access to configure edges with OpenID Connect.
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:35]:
```suggestion
- An ngrok enterprise account with an access token or admin access to configure edges with OpenID Connect.
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+      ![tunnel confirmed](img/descope-3.png)
+
+1.  In the test edge, copy the **endpoint URL**. (You use this URL to test the Descope Authentication)
+    ![tunnel url](img/descope-4.png)
+
+## Step 3: Test the integration {#test-sso}
+
+1. In your browser, launch an incognito window.
+
+1. Access your ngrok tunnel (i.e., `https://descope-sso-test.ngrok.io` or using the copied endpoint URL).
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:157]:
```suggestion
1. Access your ngrok endpoint (i.e., `https://descope-sso-test.ngrok.io` or using the copied URL).
```

File: docs/integrations/descope/sso-saml.mdx [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+     ![tunnel confirmed](img/descope-3.png)
+
+1. In the test edge, copy the **endpoint URL**. (you use this URL to test the Descope Authentication)
+   ![tunnel url](img/descope-4.png)
+
+## Test the integration {#test-sso}
+
+1. In your browser, launch an incognito window.
+
+1. Access your ngrok tunnel (i.e., `https://descope-sso-test.ngrok.app`. Use the copied endpoint URL).
```
Comment (russorat) [inline docs/integrations/descope/sso-saml.mdx:126]:
```suggestion
1. Access your ngrok endpoint (i.e., `https://descope-sso-test.ngrok.app` or use the copied URL).
```

**Comment by russorat** [comment]
sorry for the delay. i'm cleaning this up and will merge.

**Comment by gaokevin1** [comment]
No worries thanks so much!

File: docs/integrations/descope/sso-oidc.mdx [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+
+   ```bash
+   ngrok http 3000 --oidc=DESCOPE_ISSUER_URL \
+   --oidc-client-id=DESCOPE_CLIENT_ID \
+   --oidc-client-secret=DESCOPE_CLIENT_SECRET \
+   ```
+
+   **Note**: Replace the following with values:
+
+   - DESCOPE_ISSUER_URL: This is your Issuer URL that you copied from the **Applications** page earlier.
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:91]:
```suggestion
   - `DESCOPE_ISSUER_URL`: This is your Issuer URL that you copied from the **Applications** page earlier.
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+   - DESCOPE_CLIENT_ID: This is your Descope Project ID, which can be found under [Project Settings](https://app.descope.com/settings/project).
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:92]:
```suggestion
   - `DESCOPE_CLIENT_ID`: This is your Descope Project ID, which can be found under [Project Settings](https://app.descope.com/settings/project).
```

File: docs/integrations/descope/sso-oidc.mdx [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+   - DESCOPE_CLIENT_SECRET: The is the Descope Access Key that you created from earlier.
```
Comment (russorat) [inline docs/integrations/descope/sso-oidc.mdx:93]:
```suggestion
   - `DESCOPE_CLIENT_SECRET`: The is the Descope Access Key that you created from earlier.
```

File: docs/integrations/descope/sso-saml.mdx [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+### **Step 3**: Configure Descope {#configure-descope}
+
+1. Back in the Descope Console, under your **ngrok** application, select **Enter the connection details manually** under **Service Provider (SP)**.
+
+1. Enter the following values in the input fields and save the configuration.
+   ![SP configuration in Descope](img/descope-7.png)
+
+   **Note**: Replace the following with the values copied on previous steps:
+
+   - NGROK_ACS_URL: The value of the ngrok **Entity ID** copied from the edge SAML configuration.
```
Comment (russorat) [inline docs/integrations/descope/sso-saml.mdx:93]:
```suggestion
   - `NGROK_ACS_URL`: The value of the ngrok **Entity ID** copied from the edge SAML configuration.
```

File: docs/integrations/descope/sso-saml.mdx [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+   - NGROK_ENTITY_ID_URL: The value of the ngrok **ACS URL** copied from the edge SAML configuration.
```
Comment (russorat) [inline docs/integrations/descope/sso-saml.mdx:94]:
```suggestion
   - `NGROK_ENTITY_ID_URL`: The value of the ngrok **ACS URL** copied from the edge SAML configuration.
```

File: docs/integrations/descope/sso-saml.mdx [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+
+### **Step 5**: Start a Tunnel {#start-tunnel}
+
+1. Access the [ngrok edges page](https://dashboard.ngrok.com/cloud-edge/edges), click your edge, and then click **Start a tunnel**.
+
+:::tip Note
+For this step, we assume you have an app running locally (i.e. on localhost:3000) with the ngrok client installed.
```
Comment (russorat) [inline docs/integrations/descope/sso-saml.mdx:101]:
```suggestion
For this step, we assume you have an app running locally (i.e. on `localhost:3000`) with the ngrok client installed.
```

**Comment by gaokevin1** [comment]
@russorat could you please merge this?

**Review by russorat — APPROVED** [review]
lgtm

**Pull request merged**